client/extensions/woocommerce/state/sites/orders/activity-log/selectors.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { areOrderNotesLoaded, areOrderNotesLoading, getOrderNotes } from '../notes/selectors';
+import {
+	isLoaded as areShippingLabelsLoaded,
+	isFetching as areShippingLabelsLoading,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors'
+
+export const EVENT_TYPES = {
+	INTERNAL_NOTE: 'INTERNAL_NOTE',
+	CUSTOMER_NOTE: 'CUSTOMER_NOTE',
+	LABEL_PURCHASED: 'LABEL_PURCHASED',
+	LABEL_REFUND_REQUESTED: 'LABEL_REFUND_REQUESTED',
+	LABEL_REFUND_COMPLETED: 'LABEL_REFUND_COMPLETED',
+	LABEL_REFUND_REJECTED: 'LABEL_REFUND_REJECTED',
+};
+
+export const isActivityLogLoaded = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	return areOrderNotesLoaded( state, orderId, siteId ) &&
+		( ! config.isEnabled( 'woocommerce/extension-wcservices' ) || areShippingLabelsLoaded( state, orderId, siteId ) );
+};
+
+export const isActivityLogLoading = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	return areOrderNotesLoading( state, orderId, siteId ) &&
+		( ! config.isEnabled( 'woocommerce/extension-wcservices' ) || areShippingLabelsLoading( state, orderId, siteId ) );
+};
+
+export const getActivityLogEvents = ( state, orderId, siteId = getSelectedSiteId( state ) ) => {
+	const events = getOrderNotes( state, orderId, siteId ).map( note => ( {
+		key: note.id,
+		type: note.customer_note ? EVENT_TYPES.CUSTOMER_NOTE : EVENT_TYPES.INTERNAL_NOTE,
+		timestamp: new Date( note.date_created_gmt + 'Z' ).getTime(),
+		content: note.note,
+	} ) );
+
+	if ( config.isEnabled( 'woocommerce/extension-wcservices' ) ) {
+		const labels = get( state, [ 'extensions', 'woocommerce', 'woocommerceServices', siteId, 'shippingLabel', orderId, 'labels' ], [] );
```

Comment:
I feel like we should we use the [WCS selector](https://github.com/Automattic/wp-calypso/pull/17676/files#diff-111b3d32a30c9153b2b5a2bf4fb01035R10) for the shipping label state — what do you think? (FYI there's also one for the [labels themselves](https://github.com/Automattic/wp-calypso/pull/17757/files#diff-111b3d32a30c9153b2b5a2bf4fb01035R29) that I added in #17757)

Reply:
Good! I'll steal the selector you added on #17757. You may have some trouble rebasing though :(

Reply:
Nope— no conflict